### PR TITLE
Use VSC Notebook API in stable

### PR DESCRIPTION
**Pull request by DonJayamanne**
For #4488

### CODE REVIEW

File: src/client/extensionActivation.ts [hunk 1]
```diff
@@ -83,25 +83,12 @@ async function activateLegacy(
...
-    setLoggingLevel(configuration.getSettings().logging.level);
-
     // Load the two data science experiments that we need to register types
     // Await here to keep the register method sync
     const experimentService = serviceContainer.get<IExperimentService>(IExperimentService);
     experimentService.logExperiments();
-    let useVSCodeNotebookAPI = await experimentService.inExperiment(Experiments.NativeNotebook);
+    let useVSCodeNotebookAPI =
+        env.appName.includes('Insider') || (await experimentService.inExperiment(Experiments.NativeNotebook));
     let inCustomEditorApiExperiment = await experimentService.inExperiment(Experiments.CustomEditor);
```
Comment (rchiodo) [inline src/client/extensionActivation.ts:92]: I think inCustomEditorApiExperiment should be false if nativeEditor is true.

Comment (rchiodo) [reply src/client/extensionActivation.ts:92]: Maybe in the experiment service level? So nothing can have both of them turned on at the same time.

**Review by DavidKutu — CHANGES_REQUESTED** [review]
We should make sure a user can't be in both experiments at once.
Or if they're in both, useVSCodeNotebookAPI  should stay true and inCustomEditorApiExperiment  should go to false.

File: src/client/extensionActivation.ts (hunk 1 above)
Comment (DonJayamanne) [reply src/client/extensionActivation.ts:92]: This is done further below in that code.

File: src/client/extensionActivation.ts [hunk 2]
```diff
@@ -112,6 +99,20 @@ async function activateLegacy(
         inCustomEditorApiExperiment = false;
```
Comment (DonJayamanne) [inline src/client/extensionActivation.ts:99]: Please chekc the line above this @rchiodo , if both are enabled, then native notebook is false with a message.

Comment (DonJayamanne) [reply src/client/extensionActivation.ts:99]: Please let me know if this addresses your concern, Check line `111`.  @rchiodo

**Review by DonJayamanne — COMMENTED** [review]
Accidental comments

File: src/client/extensionActivation.ts (hunk 2 above)
Comment (rchiodo) [reply src/client/extensionActivation.ts:99]: Sorry but that seems incorrect too. It's possible to be in both now without any work on the user. I don't think they'd know what to do.

Comment (joyceerhl) [reply src/client/extensionActivation.ts:99]: > I don't think they'd know what to do.
Can we give users an option on the error message to 'Use Custom Editor' or 'Use Native Notebooks', and then opt them into just the one that they select?

Alternatively we could give precedence to whatever the user has in their settings.json, i.e. if the user has opted into custom editor in their settings.json, and the experiment service has opted them into native notebooks, we should opt them out of native notebooks to prevent the conflict since they've explicitly asked to be in custom editor. i.e. manually configured settings should have priority over our experiment.

Comment (DonJayamanne) [reply src/client/extensionActivation.ts:99]: Sure, but if the user is opting into something manually then they should also know how to opt out. Not sure its worth the effort, will check with PMs.

Comment (DonJayamanne) [reply src/client/extensionActivation.ts:99]: Upon further review, I do'nt agree wit the sugestion:
>  'Use Custom Editor' or 'Use Native Notebooks', and then opt them into just the one that they select?

The user already added them selves manually, hence they can opt out.
This should be a rare case as we haven't actively asked users to opt into the experiment as we had a few issues in custom editor that made is unusable.

> we should opt them out of native notebooks to prevent the conflict since they've explicitly asked to be in custom editor. i.e. manually configured settings should have priority over our experiment.

Will bring on teams, i think we will have to do this.

Comment (DonJayamanne) [reply src/client/extensionActivation.ts:99]: Will add new custom editor, see discussion here https://teams.microsoft.com/l/message/19:9b163ad1e1ed4c49938d9692be1ce751@thread.skype/1611862176803?tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47&groupId=6182cd1a-acf2-40e6-8b08-07bd25990df8&parentMessageId=1611784871932&teamName=Python%20for%20VS%20Code&channelName=Data%20Science&createdTime=1611862176803

Comment (DonJayamanne) [reply src/client/extensionActivation.ts:99]: @rchiodo @DavidKutu pleasee re-review

File: src/client/extensionActivation.ts [hunk 3]
```diff
@@ -83,25 +83,12 @@ async function activateLegacy(
     const fs = serviceManager.get<IFileSystem>(IFileSystem);
     await setExtensionInstallTelemetryProperties(fs);
 
-    const applicationEnv = serviceManager.get<IApplicationEnvironment>(IApplicationEnvironment);
```
Comment (rchiodo) [inline src/client/extensionActivation.ts:86]: Why did this stuff go away? Seems unrelated?

Comment (rchiodo) [reply src/client/extensionActivation.ts:86]: Nevermind. It just moved.

File: src/client/extensionActivation.ts (hunk 2 above)
Comment (rchiodo) [reply src/client/extensionActivation.ts:99]: You didn't change this? Not sure what the difference is. It's still possible to be in both experiments at the same time. Or are you saying it's not possible unless the user explicitly does it? Because we're changing the custom editor experiment in control tower?

However when we change the 'control tower' value for customEditor, it should use the same optIn/optOutFrom value in our package.json, right? So that users that already using don't have to change their setting? 

Regardless of that, this code still seems wrong. 

User gets put in native experiment. 
They're already opting into custom editor 

This error comes up. 

They don't know why. What are they supposed to do?

**Review by rchiodo — CHANGES_REQUESTED** [review]
⏲️

File: src/client/extensionActivation.ts (hunk 2 above)
Comment (DonJayamanne) [reply src/client/extensionActivation.ts:99]: > Or are you saying it's not possible unless the user explicitly does it?

Yes this was my assumption. The experiments should be mutually exclusive else the whole experiment framework seems busted. 
@claudiaregio Please can you confirm that users in CustomEditor will not end up in NativeNotebooks as well

Comment (DonJayamanne) [reply src/client/extensionActivation.ts:99]: > They don't know why. What are they supposed to do?

This is pre-existing code and we were always displaying errors to the user, I've only changed this PR to ensure we can use NativeNotebook in stable.

Comment (rchiodo) [reply src/client/extensionActivation.ts:99]: Even if control tower makes it impossible, it's still possible for a user to be in one manually and then end up in the other one by 'accident'.

Comment (DonJayamanne) [reply src/client/extensionActivation.ts:99]: > However when we change the 'control tower' value for customEditor, it should use the same optIn/optOutFrom value in our package.json, right? So that users that already using don't have to change their setting?

@greazer suggested we create a new experiment for Custom Editor.
As this PR focuses on just enabling Native Notebooks in stable, I"m not going to fix custom editror & the like here. Theres a separate issue to address custom editor (experiment) issues  https://github.com/microsoft/vscode-jupyter/issues/4561.

Comment (DonJayamanne) [reply src/client/extensionActivation.ts:99]: Here's the existing code that prevents users from having both experiments turned on:
```typescript
    if (useVSCodeNotebookAPI && inCustomEditorApiExperiment) {
        const appShell = serviceContainer.get<IApplicationShell>(IApplicationShell);
        appShell.showErrorMessage(localize.DataScience.illegalEditorConfig()).then(noop, noop);
        useVSCodeNotebookAPI = false;
        inCustomEditorApiExperiment = false;
```

**Review by DavidKutu — APPROVED** [review]

File: src/client/extensionActivation.ts (hunk 2 above)
Comment (rchiodo) [reply src/client/extensionActivation.ts:99]: Okay so that error message may come up but you'll fix it elsewhere? That's good for me.

**Review by rchiodo — APPROVED** [review]
